Modules/Challenges/src/Challenges.API.Client/API/ChallengesClient.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Net;
+using Challenges.API.Client.Models;
+using RestSharp;
+
+namespace Challenges.API.Client.API;
+public class ChallengesClient
+{
+    private readonly RestClient _client;
+
+    public ChallengesClient()
+    {
+        _client = new RestClient("http://localhost:5000");
+
+        ServicePointManager.ServerCertificateValidationCallback +=
+                (sender, cert, chain, sslPolicyErrors) => true;
+    }
+
+    public async Task<RestResponse<AccessTokenResponse>> GetAccessTokenAsync(AuthenticationRequest authenticationRequest)
+    {
+        var request = new RestRequest("/connect/token", Method.Post);
+        request.AddParameter("grant_type", "password");
+        request.AddParameter("client_id", authenticationRequest.ClientId);
+        request.AddParameter("client_secret", authenticationRequest.ClientSecret);
+        request.AddParameter("username", authenticationRequest.Username);
+        request.AddParameter("password", authenticationRequest.Password);
+
+        return await _client.ExecuteAsync<AccessTokenResponse>(request);
```

Comment:
Did you test that? Because I wouldn't expect this to work. The /connect/token endpoint expects the content type `application/x-www-form-urlencoded`, and I don't think that the client uses that by default.